examples/c/domains/src/subscriber.c

```diff
@@ -91,32 +91,32 @@ int main(int argc, char** argv) {
     }
 
     // create subscriber
-    iox2_port_factory_pub_sub_ref_h ref_service = iox2_cast_port_factory_pub_sub_ref_h(service);
+    iox2_port_factory_pub_sub_h_mut ref_service = iox2_cast_port_factory_pub_sub_h_mut(service);
     iox2_port_factory_subscriber_builder_h subscriber_builder =
         iox2_port_factory_pub_sub_subscriber_builder(ref_service, NULL);
     iox2_subscriber_h subscriber = NULL;
     if (iox2_port_factory_subscriber_builder_create(subscriber_builder, NULL, &subscriber) != IOX2_OK) {
         printf("Unable to create subscriber!\n");
         goto drop_service;
     }
-    iox2_subscriber_ref_h subscriber_ref = iox2_cast_subscriber_ref_h(subscriber);
+    iox2_subscriber_h_mut subscriber_mut = iox2_cast_subscriber_h_mut(subscriber);
 
     uint64_t counter = 0;
     printf("subscribed to: [domain: \"%s\", service: \"%s\"]\n", argv[1], argv[2]);
-    while (iox2_node_wait(node_ref_handle, 1, 0) == iox2_node_event_e_TICK) {
+    while (iox2_node_wait(node_handle_mut, 1, 0) == iox2_node_event_e_TICK) {
         counter += 1;
 
         // receive sample
         iox2_sample_h sample = NULL;
-        if (iox2_subscriber_receive(subscriber_ref, NULL, &sample) != IOX2_OK) {
+        if (iox2_subscriber_receive(subscriber_mut, NULL, &sample) != IOX2_OK) {
             printf("Failed to receive sample\n");
             goto drop_subscriber;
         }
 
         if (sample != NULL) {
-            iox2_sample_ref_h sample_ref = iox2_cast_sample_ref_h(sample);
+            iox2_sample_h_mut sample_mut = iox2_cast_sample_h_mut(sample);
```

Comment:
This is also not nice. Could become
```suggestion
            iox2_sample_h_ref sample_ref = iox2_cast_sample_h_ref(sample);
```

Reply:
If not mutable, agree.